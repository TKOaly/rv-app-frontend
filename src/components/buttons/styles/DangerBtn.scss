@import "vars";

/*
  Danger buttons
  */

.danger {
  color: $danger-button-bgcolor;
  -webkit-transition: background-color 0.1s; /* Safari */
  transition: background-color 0.1s;
}

.danger-hover:hover {
  color: $danger-button-textcolor;
  background-color: $danger-button-bgcolor;
  border-color: $danger-button-bgcolor;
  cursor: pointer;
}

.danger-fill {
  color: $danger-button-textcolor;
  background-color: $danger-button-bgcolor;
  border-color: $danger-button-bgcolor;
  -webkit-transition: background-color 0.1s; /* Safari */
  transition: background-color 0.1s;
}

.danger-fill-hover:hover {
  color: $danger-button-bgcolor;
  background-color: transparent;
  border-color: $danger-button-bgcolor;
  cursor: pointer;
}
