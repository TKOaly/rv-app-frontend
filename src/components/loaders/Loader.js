import './styles/Loader.css';
import React from 'react';

const Loader = () => <div className="loader" />;

export default Loader;
