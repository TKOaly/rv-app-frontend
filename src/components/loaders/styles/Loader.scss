@import "vars";

.loader {
  border: 2px solid transparent;
  border-top: $loader-width solid $loader-color;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  animation: spin 0.5s linear infinite;
  margin-left: auto;
  margin-right: auto;
}
@keyframes spin {
  
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
