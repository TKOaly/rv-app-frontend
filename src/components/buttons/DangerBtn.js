import './styles/Button.scss';
import './styles/DangerBtn.scss';
import Loader from '../loaders/Loader';
import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

const DangerBtn = ({ onClick, children, fill, hover, loader, ...props }) => {
    return (
        <button
            {...props}
            onClick={onClick}
            className={classNames('btn', {
                'danger-fill-hover': fill && hover,
                'danger-hover': !fill && hover,
                danger: !fill,
                'danger-fill': fill
            })}
        >
            {!loader ? <span className="btnContent">{children}</span> : <Loader />}
        </button>
    );
};

DangerBtn.propTypes = {
    /** Click handler. */
    onClick: PropTypes.func,
    /** Button fill. If set to `true`, the button's colors will "invert". */
    fill: PropTypes.bool,
    /** Hover effect. If set to `true`, the button will have a hover effect. */
    hover: PropTypes.bool,
    /** Loader effect. If set to `true`, will show a CSS loader instead of button text. */
    loader: PropTypes.bool
};

DangerBtn.defaultProps = {
    onClick: null,
    fill: false,
    hover: false,
    loader: false
};

export default DangerBtn;
