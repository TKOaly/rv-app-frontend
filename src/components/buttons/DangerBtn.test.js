import { shallow } from 'enzyme';
import DangerBtn from './DangerBtn';
import Loader from '../loaders/Loader';
import React from 'react';
import renderer from 'react-test-renderer';

describe.only('<DangerBtn />', () => {
    it('renders text correctly', () => {
        const data = {
            text: 'Hello world'
        };

        const dangerBtnComponent = shallow(<DangerBtn>{data.text}</DangerBtn>);
        console.log(dangerBtnComponent.debug());
        const contentDiv = dangerBtnComponent.find('.btnContent');
        expect(contentDiv.text()).toContain(data.text);

        const cmpnt = renderer.create(<DangerBtn>{data.text}</DangerBtn>).toJSON();
        expect(cmpnt).toMatchSnapshot();
    });

    it('shows loader correctly', () => {
        const dangerBtnComponent = shallow(<DangerBtn loader />);
        console.log(dangerBtnComponent.debug());
        expect(dangerBtnComponent.contains(<Loader />)).toBe(true);
        const cmpnt = renderer.create(<DangerBtn loader />).toJSON();
        expect(cmpnt).toMatchSnapshot();
    });

    it('onClick event works correctly', () => {
        const data = {
            text: 'Hello world'
        };

        const mockHandler = jest.fn();

        const dangerBtnComponent = shallow(<DangerBtn onClick={mockHandler} />);

        const button = dangerBtnComponent.find('button');

        const cmpnt = renderer.create(<DangerBtn onClick={mockHandler} />).toJSON();
        expect(cmpnt).toMatchSnapshot();

        expect(mockHandler.mock.calls.length).toBe(0);
        button.simulate('click');
        expect(mockHandler.mock.calls.length).toBe(1);
    });
});
