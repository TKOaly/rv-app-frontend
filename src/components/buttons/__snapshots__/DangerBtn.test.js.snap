// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DangerBtn /> onClick event works correctly 1`] = `
<button
  className="btn danger"
  onClick={[MockFunction]}
>
  <span
    className="btnContent"
  />
</button>
`;

exports[`<DangerBtn /> renders text correctly 1`] = `
<button
  className="btn danger"
  onClick={null}
>
  <span
    className="btnContent"
  >
    Hello world
  </span>
</button>
`;

exports[`<DangerBtn /> shows loader correctly 1`] = `
<button
  className="btn danger"
  onClick={null}
>
  <div
    className="loader"
  />
</button>
`;
